### DISCUSSION TITLE
Is this the correct way to add server middleware to a module?

### DISCUSSION THREAD

**Discussion by andyjamesn** [nuxt/nuxt#17503] [Questions] [answered]
I have a Nuxt module which has a server and I am trying to define the server in the module.ts

I initially used addServerMiddleware but noticed it was deprecated and replaced recently with serverHandlers. 

There is very little documentation out there on how to implement this so I looked at the code to try figure it out and this is what I have come up with and it is working fine.

Is this the correct way to do this? Here is the main code for my module

```javascript
const runtimeDir = fileURLToPath(new URL('./runtime', import.meta.url))
nuxt.options.build.transpile.push(runtimeDir)
addPlugin(resolve(runtimeDir, 'plugin'))
nuxt.hook('autoImports:dirs', (dirs) => {
  dirs.push(resolve(runtimeDir, 'composables'))
})

// THIS IS HOW I ADDED THE SERVER HANDLER
nuxt.options.serverHandlers.push({
  route: '/api/hello',
  handler: resolve(runtimeDir, 'server/api/hello')
})
```

**Comment by wobsoriano** [comment, marked as answer]
That's correct.

You can also use `addServerHandler` from `@nuxt/kit` (if you're developing a module):

```ts
import { addServerHandler } from '@nuxt/kit'

addServerHandler({
  route: '/api/hello',
  handler: resolve(runtimeDir, 'server/api/hello')
})
```
